src/Connect.js

```diff
@@ -1,68 +1,201 @@
    * @param       {String}            opts.infuraApiKey      Infura API Key (register here http://infura.io/register.html)
    * @param       {Function}          opts.topicFactory      function which generates topics and deals with requests and response
    * @param       {Function}          opts.uriHandler        default function to consume generated URIs for requests, can be used to display QR codes or other custom UX
    * @param       {Function}          opts.mobileUriHandler  default function to consume generated URIs for requests on mobile
    * @param       {Function}          opts.closeUriHandler   default function called after a request receives a response, can be to close QR codes or other custom UX
+   * @param      {String}            opts.accountType       Ethereum account type: "general", "segregated", "keypair", "devicekey" or "none"
+   * @param       {Object}            opts.ethrConfig        Configuration object for ethr did resolver. See [ethr-did-resolver](https://github.com/uport-project/ethr-did-resolver)
    * @return      {Connect}                                  self
    */
-
   constructor (appName, opts = {}) {
-    super(appName, opts)
-    this.uriHandler = opts.uriHandler || openQr
-    this.mobileUriHandler = opts.mobileUriHandler || mobileUriHandler
-    this.closeUriHandler = opts.closeUriHandler || (this.uriHandler === openQr ? closeQr : undefined)
+    // Config
+    this.appName = appName || 'uport-connect-app'
+    this.infuraApiKey = opts.infuraApiKey || this.appName.replace(/\W+/g, '-')  //Not used right now
+    this.provider = opts.provider || new HttpProvider(this.network.rpcUrl)
+    this.accountType = opts.accountType
+    this.isOnMobile = opts.isMobile === undefined ? isMobile() : opts.isMobile
+    this.network = network.config.network(opts.network)
+    // State
+    this.address = null
+    this.firstReq = true
+    // Transports
+    this.transport = opts.transport || connectTransport
+    this.mobileTransport = opts.mobileTransport || transport.url.send()
+    this.PubSub = PubSub
+    this.onloadResponse = transport.url.getResponse()
+    transport.url.listenResponse((err, res) => {
+      if (res) this.PubSub.publish(res.id, {res: res.res, data: res.data}) // TODO pass errors
+    })
+    // TODO need a nicer way configure resolvers, lines below should be temporary
+    // TODO would be nice to have some uncoupled funct for parsing/verifying in uport-core-js rather than require config a credential object, nice to require aud either here
+    UportDIDResolver(UportLite({ networks: network.config.networkToNetworkSet(this.network) }))
+    EthrDIDResolver(opts.ethrConfig || {})
+    this.verifyResponse = (res) => {
+      const decodedToken = decodeJWT(res).payload
```

Comment:
Instead of all of this use `credentials.authenticate()`. What you're doing here is breaking the `aud` model